packages/playwright-core/src/server/transport.ts

```diff
@@ -90,13 +91,26 @@ export class WebSocketTransport implements ConnectionTransport {
 
   constructor(progress: Progress|undefined, url: string, headers?: { [key: string]: string; }, followRedirects?: boolean) {
     this.wsEndpoint = url;
+
+    const endpointURL = new URL(url);
+    const proxyURL = url.startsWith('ws:') ? getProxyForUrl(`http://${endpointURL.hostname}`) : getProxyForUrl(`https://${endpointURL.hostname}`);
+    let agent: Agent | undefined = undefined;
+    if (proxyURL) {
+      const parsedProxyURL = new URL(proxyURL);
+      (parsedProxyURL as any).secureProxy = parsedProxyURL.protocol === `https:`;
+
+      agent = new HttpsProxyAgent(parsedProxyURL);
+    }
+
     this._ws = new ws(url, [], {
       perMessageDeflate: false,
       maxPayload: 256 * 1024 * 1024, // 256Mb,
       // Prevent internal http client error when passing negative timeout.
       handshakeTimeout: Math.max(progress?.timeUntilDeadline() ?? 30_000, 1),
       headers,
       followRedirects,
+      agent: agent,
+      rejectUnauthorized: false,
```

Comment:
Why did this change?

Reply:
Because you've added the same line in [here](https://github.com/microsoft/playwright/blob/b7a49fb3714705c3b487179012c778c5d2922391/packages/playwright-core/src/common/netUtils.ts#L66)

Reply:
I don't think ignoring TLS in this case makes sense, let's remove this.